### ISSUE TITLE
set-style API - descriptors cannot be empty

### ISSUE THREAD

**Issue by bcollard** [solo-io/gloo#4279]
**Describe the bug**
WIth the Set-Style API, one cannot create RLC having no _descriptors_.
With the Envoy style, it is possible.

The use case is a deployment with a single VirtualService, pointing to a single RLC. This RLC will be modified dynamically by a third-party app. This app will set limits based upon an external CMDB.

**To Reproduce**
Envoy RL - no rateLimit rule (like 1 hit / minute) - **this is accepted**
```yaml
apiVersion: ratelimit.solo.io/v1alpha1
kind: RateLimitConfig
metadata:
  name: voice-envoy
  namespace: gloo-system
spec:
  raw:
    descriptors:
    - alwaysApply: false
      key: api
      value: sms
      weight: 0
    rateLimits:
    - actions:
      - request_headers:
          descriptor_key: api
          header_name: x-api
```

SetStyle API - no rateLimit rule (like 1 hit / minute) - **this is not accepted**:
```yaml
apiVersion: ratelimit.solo.io/v1alpha1
kind: RateLimitConfig
metadata:
    name: voice
    namespace: gloo-system
spec:
    raw:
        rateLimits:
        -   setActions:
            -   requestHeaders:
                    descriptorKey: api
                    headerName: x-api
        setDescriptors: []
```

**Expected behavior**
Get the same user experience with the SetStyle API.

**Additional context**
- Gloo Edge version: 1.6.9

**Comment by kdorosh** [comment]
Both config will be accepted unless referenced from a virtual service, e.g.
```yaml
apiVersion: gateway.solo.io/v1
kind: VirtualService
metadata:
  name: httpbin
  namespace: gloo-system
spec:
  virtualHost:
    domains:
      - '*'
    routes:
      - matchers:
          - prefix: /
        routeAction:
            single:
              upstream:
                name: default-httpbin-8000
                namespace: gloo-system
    options:
      rateLimitConfigs:
        refs:
        - name: voice
          namespace: gloo-system
```

**Comment by kdorosh** [comment]
Not sure what the inconsistency here is. This config is accepted:

```yaml
apiVersion: ratelimit.solo.io/v1alpha1
kind: RateLimitConfig
metadata:
    name: voice
    namespace: gloo-system
spec:
    raw:
        rateLimits:
        -   setActions:
            -   requestHeaders:
                    descriptorKey: api
                    headerName: x-api
        setDescriptors:
         - alwaysApply: false
           simpleDescriptors:
           - key: api
             value: sms
```

**Comment by kdorosh** [comment]
Ah, I see the inconsistency. The original yaml as example is wrong. The inconsistency is that
```yaml
apiVersion: ratelimit.solo.io/v1alpha1
kind: RateLimitConfig
metadata:
  name: voice-envoy
  namespace: gloo-system
spec:
  raw:
    descriptors: []
    rateLimits:
    - actions:
      - request_headers:
          descriptor_key: api
          header_name: x-api
```

is accepted but

```yaml
apiVersion: ratelimit.solo.io/v1alpha1
kind: RateLimitConfig
metadata:
    name: voice
    namespace: gloo-system
spec:
    raw:
        rateLimits:
        -   setActions:
            -   requestHeaders:
                    descriptorKey: api
                    headerName: x-api
        setDescriptors: []
```

is not

**Comment by kdorosh** [comment]
The following are both accepted, even though they don't have rate limits defined on the `descriptors` or `setDescriptors` (as designed):

```yaml
apiVersion: ratelimit.solo.io/v1alpha1
kind: RateLimitConfig
metadata:
  name: voice
  namespace: gloo-system
spec:
  raw:
    descriptors:
    - alwaysApply: false
      key: api
      value: apivalue
      weight: 0
    rateLimits:
    - actions:
      - request_headers:
          descriptor_key: api
          header_name: x-api
```
and

```yaml
apiVersion: ratelimit.solo.io/v1alpha1
kind: RateLimitConfig
metadata:
    name: voice
    namespace: gloo-system
spec:
    raw:
        rateLimits:
        -   setActions:
            -   requestHeaders:
                    descriptorKey: api
                    headerName: x-api
        setDescriptors:
        - simpleDescriptors:
          - key: api
            value: apivalue
```

**Comment by kdorosh** [comment]
Update:

In the config from https://github.com/solo-io/gloo/issues/4279#issuecomment-780733019 ,

note that while both are accepted on the CRD, the earlier one does not cause any error logs in Gloo or prevent rate-limit from receiving other updates, unlike the latter. This has been reproduced and fix is going through internal ci.

**Comment by kdorosh** [comment]
Fix is out in 1.6.20
